File: day-2/part-2.js
const data = require('./data')

let numValidPasswords = 0

data.forEach(datum => {
  const [minMax, charWithSuffix, word] = datum.split(' ')
  const [minChar, maxChar] = minMax.split('-')
  const left = Number(minChar)
  const right = Number(maxChar)
  const char = charWithSuffix.replace(':', '')

  let count = 0

  if (word[left - 1] === char) {
    count += 1
  }

  if (word[right - 1] === char) {
    count += 1
  }

  if (count === 1) {
    numValidPasswords += 1
  }
})

console.log('numValidPasswords: ', numValidPasswords)

File: day-2/part-2.py
with open('data.txt') as f:
  data = f.read().splitlines()

  numValidPasswords = 0

  for datum in data:
    minMax, charWithSuffix, word = datum.split(' ')
    minChar, maxChar = minMax.split('-')
    left = int(minChar)
    right = int(maxChar)
    char = charWithSuffix.replace(':', '')

    count = 0

    if word[left - 1] == char:
      count += 1

    if word[right - 1] == char:
      count += 1

    if count == 1:
      numValidPasswords +=1

  print(numValidPasswords)